CONTRIBUTING.md

```diff
@@ -0,0 +1,174 @@
+We welcome contributions from the community. Please read the following guidelines carefully to
+maximize the chances of your PR being merged.
+
+# Communication
+
+* Before starting work on a major feature, please reach out to us via GitHub, Slack,
+  email, etc. We will make sure no one else is already working on it and ask you to open a
+  GitHub issue.
+* Small patches and bug fixes don't need prior communication.
+
+# Coding style
+
+* Coding style mirrors [Envoy's policy](https://github.com/envoyproxy/envoy/blob/master/STYLE.md)
+
+# Breaking change policy
+
+Both API and implementation stability are important to Nighthawk. Since the API is consumed by clients beyond Nighthawk, breaking changes to that are prohibited.
+
+# Submitting a PR
+
+* Fork the repo.
```

Comment:
For any verbose section, if it's the same as Envoy, just link back to the relevant anchor link in the Envoy docs. If there's a small delta, then feel free to link and state the difference.